benchmarks/src/jmh/kotlin/benchmarks/flow/FlattenMergeBenchmark.kt

```diff
@@ -0,0 +1,58 @@
+package benchmarks.flow
+
+import benchmarks.doWork
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+import org.openjdk.jmh.annotations.*
+import java.util.concurrent.ThreadLocalRandom
+import java.util.concurrent.TimeUnit
+import kotlin.math.ceil
+
+/**
+ * This benchmark can be considered as a macro benchmark for the [kotlinx.coroutines.sync.Semaphore]
```

Comment:
Semaphore is just an implementation detail, I would not say that this benchmark is for it